docs/docs/building-blocks/6-optimizers.md

```diff
@@ -38,36 +38,32 @@ All of these can be accessed via `from dspy.teleprompt import *`.
+2. **`BootstrapFewShot`**: Uses a `teacher` module (which defaults to your program) to generate complete demonstrations for every stage of your program, along with labeled examples in `trainset`. Parameters include `max_labeled_demos` (the number of demonstrations randomly selected from the `trainset`) and `max_bootstrapped_demos` (the number of additional examples generated by the `teacher`). The bootstrapping process employs the metric to validate demonstrations, including only those that pass the metric in the "compiled" prompt. Advanced: Supports using a `teacher` program that is a *different* DSPy program that has compatible structure, for harder tasks.
 
-3. **`BootstrapFewShotWithRandomSearch`**: Applies `BootstrapFewShot` several times with random search over generated demonstrations, and selects the best program.  Takes as parameter (in addition to the ones for `BootstrapFewShot`) `num_candidate_programs`: the number of random programs, each generated through `BootstrapFewShot`.  Picks a single best based on scores over the `valset` (which defaults to being the same as the `trainset`).
+3. **`BootstrapFewShotWithRandomSearch`**: Applies `BootstrapFewShot` several times with random search over generated demonstrations, and selects the best program over the optimization. Parameters mirror those of `BootstrapFewShot`, with the addition of `num_candidate_programs`, which specifies the number of random programs evaluated over the optimization, including candidates of the uncompiled program, `LabeledFewShot` optimized program, `BootstrapFewShot` compiled program with unshuffled examples and `num_candidate_programs` of `BootstrapFewShot` compiled programs with randomized example sets. 
 
-4. **`BootstrapFewShotWithOptuna`**: Applies `BootstrapFewShot` through Optuna hyperparameter optimization across demonstration sets, running trials to maximize evaluation metrics. As before, chooses the best of a candidate set of demonstration examples, but this time using Optuna to make the choice. <!--- TBQH, I don't understand how Optuna does this. As far as I can tell it simply chooses best based on multiple evaluations, rather than a single one, and mention of "hyperparameters" seems to be a red herring. ---> 
+4. **`BootstrapFewShotWithOptuna`**: Applies `BootstrapFewShot` through Optuna hyperparameter optimization across demonstration sets, running trials to maximize evaluation metrics and selecting the best demonstrations. 
```

Comment:
The thing that confuses me about this is the use of the term "hyperparameter optimization." I don't think there are any hyperparameters (the number of examples to include in the prompt, the number of samples to take from the trainset and the number of samples to generate) actually optimized here, are there? Doesn't this just optimize the choice of examples to maximize the metric? In that case, we are just optimizing parameters, not hyperparameters. That's why I said that the mention of hyperparameters was misleading.

Or am I missing something? It's hard to tell from just looking at the code in dspy what Optuna is doing, and I didn't have time to read over Optuna.

Reply:
seems like this is a matter of semantics but I think hyperparameter is fine since each demonstration is considered and optimized by Optuna which searches over the different indices to select the combination of demonstrations that maximizes the metric.

Reply:
This isn't just semantics, because it's actively misleading.  If it's just semantics, why isn't it ok to drop the word "hyperparameter" and just have "Optuna optimization"?

The reason I am so intent on this change is that the term "hyperparameter" applies not to model parameters, but to parameters _that control the learning process_ . So in this case things like `max_labeled_demos`, `max_bootstrapped_demos`, `max_errors`, `rounds` are the hyperparameters: the set of demos are the things that are learned, the parameters.

Putting "hyperparameter" in here misleads the reader, who is led to expect optimization of the learning _process_ not optimization of the learning _product_.